examples/_HeatDiffusion/ExampleADHeatDiffusion.jl

```diff
@@ -0,0 +1,359 @@
+using StagFDTools, StagFDTools.Poisson, ExtendableSparse, StaticArrays, LinearAlgebra, Statistics, UnPack, Plots, ExactFieldSolutions
+using TimerOutputs
+# using Enzyme
+using ForwardDiff, Enzyme  # AD backends you want to use 
+
+######
+
+# This function computes the residuals of the 2D Poisson equation using a 5-point stencil
+# qx = -k*du_dx
+# qy = -k*du_dy
+# dqx_dx + dqy_dy = s
+# where qx and qy are components of a flux vector (q) in 2D and s is a source term
+# The 5-point stencil looks like this:
+#      |           |
+#      |     uN    |
+#      |           |
+#----- x -- qyN -- x -----
+#      |           |
+# uW  qxW    uC   qxE  uE
+#      |           |
+#----- x -- qyS -- x -----
+#      |           |
+#      |     uS    |
+#      |           |
+
+function Diffusion2D(u_loc, k, s, type_loc, bcv_loc, Δ, u0, ρ, cp)
+
+    # u_loc is 3*3 matrix containing the current values of u for the whole stencil
+    #             0   uN  0
+    #     u_loc = uW  uC  uE
+    #             0   uS  0
+    # Therefore u_loc[2,2] is the current value of uC
+    uC       = u_loc[2,2]
+
+    # Boundary conditions need to be applied on every boundaries :D
+    # Here we define the values of the ghost nodes. For example at the west side uW needs to be defined  
+    # For example, to set a Dirichlet values, we say: 1/2*(uW + uC) = u_BC, hence uW = 2*u_BC - uC
+    # West
+    if type_loc[1,2] === :Dirichlet
+        uW = 2*bcv_loc[1,2] - u_loc[2,2]
+    elseif type_loc[1,2] === :Neumann
+        uW = Δ.x*bcv_loc[1,2] + u_loc[2,2]
+    elseif type_loc[1,2] === :periodic || type_loc[1,2] === :in
+        uW = u_loc[1,2] 
+    end
```

Comment:
```suggestion
    uW = if type_loc[1,2] === :Dirichlet
        2*bcv_loc[1,2] - u_loc[2,2]
    elseif type_loc[1,2] === :Neumann
        Δ.x*bcv_loc[1,2] + u_loc[2,2]
    elseif type_loc[1,2] === :periodic || type_loc[1,2] === :in
        u_loc[1,2] 
    end
```
This is an alternative syntax that I personally prefer in this kind of cases, but I leave it up to you.

Reply:
uh! I never saw that one. Looks a bit messy to my taste, feels weird that  the `if` statement is no more aligned with the `elseif`.

Reply:
Any `if`, `elseif`, and `else` block actually return their last line, as in a function. I like it because you immediately see whats returned from the `if` block and there's a bit less of code around. But it's a preference of style :)